AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/src/AwsCryptographyMaterialProvidersOperations.dfy

```diff
@@ -266,10 +267,65 @@ module AwsCryptographyMaterialProvidersOperations refines AbstractAwsCryptograph
     //= type=implication
     //# If no max cache size is provided, the crypotgraphic materials cache MUST be configured to a
     //# max cache size of 1000.
-    var cache := if input.cache.Some? then
-      input.cache.value
-    else
-      Types.Default(Types.DefaultCache(entryCapacity := 1000));
+    var cmc;
+
+    if input.cache.Some? {
+      match input.cache.value {
```

Comment:
so this match is for if you provide a cache it is either 1) already initialized and potentially shared or 2) given the specs for one of the other cache types we support and we create it for them, right?

Reply:
yes